crates/proof-of-sql/src/sql/proof_exprs/decimal_scaling_cast_expr.rs

```diff
@@ -0,0 +1,96 @@
+use super::{
+    numerical_util::{
+        cast_column_to_decimal_with_scaling, try_get_scaling_factor_with_precision_and_scale,
+    },
+    DynProofExpr, ProofExpr,
+};
+use crate::{
+    base::{
+        database::{
+            try_decimal_scale_cast_types, Column, ColumnRef, ColumnType, LiteralValue, Table,
+        },
+        map::{IndexMap, IndexSet},
+        proof::{PlaceholderResult, ProofError},
+        scalar::Scalar,
+    },
+    sql::proof::{FinalRoundBuilder, VerificationBuilder},
+};
+use alloc::boxed::Box;
+use bumpalo::Bump;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub struct DecimalScalingCastExpr {
+    from_expr: Box<DynProofExpr>,
+    to_type: ColumnType,
+}
+
+impl DecimalScalingCastExpr {
+    /// Creates a new `CastExpr`
+    pub fn new(from_expr: Box<DynProofExpr>, to_type: ColumnType) -> Self {
+        Self { from_expr, to_type }
+    }
+}
+
+impl ProofExpr for DecimalScalingCastExpr {
+    fn data_type(&self) -> ColumnType {
+        try_decimal_scale_cast_types(self.from_expr.data_type(), self.to_type)
+            .expect("Failed to cast column type");
+        self.to_type
```

Comment:
Or maybe we should simply use `self.to_type`? The type check should happen elsewhere.